src/components/Form/FormProvider.tsx

```diff
@@ -0,0 +1,360 @@
+import lodashIsEqual from 'lodash/isEqual';
+import type {ForwardedRef, MutableRefObject, ReactNode} from 'react';
+import React, {createRef, forwardRef, useCallback, useImperativeHandle, useMemo, useRef, useState} from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import * as ValidationUtils from '@libs/ValidationUtils';
+import Visibility from '@libs/Visibility';
+import * as FormActions from '@userActions/FormActions';
+import CONST from '@src/CONST';
+import type {OnyxFormKey} from '@src/ONYXKEYS';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Form, Network} from '@src/types/onyx';
+import type {FormValueType} from '@src/types/onyx/Form';
+import type {Errors} from '@src/types/onyx/OnyxCommon';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import FormContext from './FormContext';
+import FormWrapper from './FormWrapper';
+import type {BaseInputProps, FormProps, InputRefs, OnyxFormKeyWithoutDraft, OnyxFormValues, OnyxFormValuesFields, RegisterInput, ValueTypeKey} from './types';
+
+// In order to prevent Checkbox focus loss when the user are focusing a TextInput and proceeds to toggle a CheckBox in web and mobile web.
+// 200ms delay was chosen as a result of empirical testing.
+// More details: https://github.com/Expensify/App/pull/16444#issuecomment-1482983426
+const VALIDATE_DELAY = 200;
+
+type InitialDefaultValue = false | Date | '';
```

Comment:
This type exists here as well https://github.com/Expensify/App/pull/32992/files#diff-8ee148181ee90e9cacaf864477b9f497b5f2cd8425a249e3837ad22e9f19b9e4R7